### PR TITLE
feat: add useSet hook

### PR DESCRIPTION
**Pull request by pikaatic**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -67,6 +67,7 @@ A super awesome collection of regularly used custom hooks as utils for React.
 *   [useSelectableList](https://react-hooks.org/docs/useSelectableList) - Easily select a single value from a list of values. very useful for radio buttons, select inputs  etc.
 *   [useSessionstorage](https://react-hooks.org/docs/useSessionstorage) - Session storage react hook. Easily manage session storage values.
 *   [useSessionstorageState](https://react-hooks.org/docs/useSessionstorageState) - useState but syncs with sessionstorage
+*   [useSet](https://react-hooks.org/docs/useSet) - React state hook that tracks a Set.
```
Comment (imbhargav5) [inline README.md:70]: "A React hook that tracks state in the form of a Set data structure."

File: src/hooks/useSet.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
+import { useCallback, useMemo, useState } from 'react';
+
+type StableActions<T> = {
```
[the remark this answers is not in the document]
Comment (imbhargav5) [reply src/hooks/useSet.ts:3]: We don't need `toggle`. Just `add`, `remove`, `removeAll`

[the remark this answers is not in the document]
Comment (imbhargav5) [reply src/hooks/useSet.ts:3]: Also let's make sure that `add` and `remove` can take any number of arguments. So if a set has `3,5,9,17` elements. 

`add(2,18)`, should make set `2,5, 9, 17, 18`

Same for remove.

File: src/__tests__/useSet.spec.ts [hunk 3]
```diff
@@ -0,0 +1,172 @@
+/**
+ * @jest-environment jsdom
+ */
```
Comment (imbhargav5) [inline src/__tests__/useSet.spec.ts:3]: This kind of Pull request is unacceptable @iampika . This is a direct rip off from `react-use` with minor rearrangements.

**Review by imbhargav5 — CHANGES_REQUESTED** [review]

File: src/hooks/useSet.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        (previousSet) =>
+          new Set(Array.from(previousSet).filter((index) => index !== item))
+      );
+    };
+
+    const toggle = (item: K) => {
+      setSet((previousSet) => {
+        if (previousSet.has(item))
+          return new Set(
+            Array.from(previousSet).filter((index) => index !== item)
```
[the remark this answers is not in the document]
Comment (iampika) [reply src/hooks/useSet.ts:40]: its actually a key (not an index), I will change it to key :)

File: src/hooks/useSet.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (iampika) [reply src/hooks/useSet.ts:3]: updated

File: README.md (hunk 1 above)
Comment (iampika) [reply README.md:70]: updated

File: README.md [hunk 5]
```diff
@@ -103,15 +104,15 @@ A super awesome collection of regularly used custom hooks as utils for React.
...
+import { useDidMount } from 'rooks';
 ```
 
 ## Usage
 
 ```jsx
 function App() {
   useDidMount(() => {
-    alert("mounted");
+    alert('mounted');
```
Comment (imbhargav5) [inline README.md:115]: @iampika Do you have some formatted enabled on your VSCode? Looks like we are getting unnecessary changes in the PR.

File: src/hooks/useSet.ts [hunk 6]
```diff
@@ -0,0 +1,43 @@
+import { useCallback, useState } from 'react';
+
+type Actions<T> = {
+  has: (key: T) => boolean;
+  add: (key: T) => void;
```
Comment (imbhargav5) [inline src/hooks/useSet.ts:5]: The types here need to be updated to take in multiple keys

**Review by imbhargav5 — CHANGES_REQUESTED** [review]

File: src/hooks/useSet.ts (hunk 6 above)
Comment (iampika) [reply src/hooks/useSet.ts]: updated

File: README.md (hunk 5 above)
Comment (iampika) [reply README.md]: fixed

File: src/hooks/useSet.ts [hunk 7]
```diff
@@ -0,0 +1,43 @@
+import { useCallback, useState } from "react";
+
+type Actions<T> = {
+  has: (key: T) => boolean;
+  add: (keys: T) => void;
+  remove: (keys: T) => void;
```
Comment (imbhargav5) [inline src/hooks/useSet.ts:6]: Hey @iampika This has to be `...items: K[]` right?

**Review by imbhargav5 — CHANGES_REQUESTED** [review]
